sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py

```diff
@@ -310,54 +466,62 @@ def _from_generated(cls, generated):
-        self.quota = kwargs.get('x-ms-share-quota')
-        self.access_tier = kwargs.get('x-ms-access-tier')
+    """Share's properties class."""
+
+    name: str
+    """The name of the share."""
+    last_modified: "datetime"
+    """A datetime object representing the last time the share was modified."""
+    etag: str
+    """The ETag contains a value that you can use to perform operations conditionally."""
+    quota: int
+    """The allocated quota."""
+    access_tier: str
+    """The share's access tier.'"""
+    next_allowed_quota_downgrade_time: Optional[str] = None
+    """The share's next allowed quota downgrade time."""
+    metadata: Dict[str, str]
+    """Name-value pairs associate with the share as metadata."""
+    snapshot: Optional[str] = None
+    """Snapshot of the share."""
+    deleted: Optional[bool] = None
+    """Whether this share was deleted. 
+        This is a service returned value, and the value will be set when list shared including deleted ones."""
+    deleted_time: Optional["datetime"] = None
+    """A datetime object representing the time at which the share was deleted.
+        This is a service returned value, and the value will be set when list shared including deleted ones."""
+    version: Optional[str] = None
+    """To indicate the version of deleted share.
+        This is a service returned value, and the value will be set when list shared including deleted ones."""
+    remaining_retention_days: Optional[int] = None
+    """The number of days that the share will be retained before being permanently deleted by the service.
+        This is a service returned value, and the value will be set when list shared including deleted ones."""
+    provisioned_egress_mbps: Optional[int] = None
+    """Provisioned egress in megabits/second. Only applicable to premium file accounts."""
+    provisioned_ingress_mbps: Optional[int] = None
+    """Provisioned ingress in megabits/second. Only applicable to premium file accounts."""
+    provisioned_iops: Optional[int] = None
+    """Provisioned input/output operators per second (iops). Only applicable to premium file accounts."""
+    provisioned_bandwidth: Optional[int] = None
+    """Provisioned bandwidth in megabits/second. Only applicable to premium file accounts."""
+    lease: LeaseProperties
+    """Share lease properties."""
+    protocols: Optional[List[str]] = None
+    """Indicates the protocols enabled on the share. The protocol can be either SMB or NFS."""
+    root_squash: Optional[Union["ShareRootSquash", str]] = None
+    """Possible values include: 'NoRootSquash', 'RootSquash', 'AllSquash'."""
+    enable_snapshot_virtual_directory_access: Optional[bool] = None
+    """Specifies whether the snapshot virtual directory should be accessible at the root of the share
+        mount point when NFS is enabled. if not specified, the default is True."""
+
+    def __init__(self, **kwargs: Any) -> None:
+        self.name = None  # type: ignore [assignment]
```

Comment:
Even though `name` is `None` here, I believe `name` is still of the type `str` since it is populated later. Making this `name` `Optional` will cause a lot of unnecessary errors throughout the code as well as large amounts of parameter type changes which might not reflect expected behaviors. It's interesting that `name` is not retrieved from `kwargs` or even one of the response headers in `x-ms-...`.

Reply:
Makes sense, seems like a case of generated code populating the field always